table/block_based_table_reader.h

```diff
@@ -717,13 +724,18 @@ class BlockBasedTableIterator : public InternalIteratorBase<TValue> {
   bool for_compaction_;
   BlockHandle prev_index_value_;
 
-  static const size_t kInitReadaheadSize = 8 * 1024;
+  // All the below fields control iterator readahead
+  static const size_t kInitAutoReadaheadSize = 8 * 1024;
   // Found that 256 KB readahead size provides the best performance, based on
-  // experiments.
-  static const size_t kMaxReadaheadSize;
-  size_t readahead_size_ = kInitReadaheadSize;
+  // experiments, for auto readahead. Experiment data is in PR #3282.
+  static const size_t kMaxAutoReadaheadSize = 256 * 1024;
+  static const int kMinNumFileReadsToStartAutoReadahead = 2;
+  size_t readahead_size_ = kInitAutoReadaheadSize;
+  size_t max_readahead_size_ = kMaxAutoReadaheadSize;
   size_t readahead_limit_ = 0;
-  int num_file_reads_ = 0;
+  int32_t start_readahead_after_num_file_reads_ =
+      kMinNumFileReadsToStartAutoReadahead;
```

Comment:
For 64-bit hosts, this is the same as int64_t. Would rather use int64 in that case.

Reply:
Sure, I can change this.

Reply:
This field has been removed entirely in the latest version.